src/rebuild-parse-cache/src/get-repos.ts

```diff
@@ -0,0 +1,173 @@
+import { GetParameterCommand, SSMClient } from '@aws-sdk/client-ssm';
+import { MongoClient } from 'mongodb';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import path from 'path';
+import { promisify } from 'util';
+import fs from 'fs';
+import {
+  APIGatewayClient,
+  GetApiKeyCommand,
+} from '@aws-sdk/client-api-gateway';
+
+const readFileAsync = promisify(fs.readFile);
+
+const REGION = 'us-east-2';
+
+async function getParameters(env: string): Promise<Record<string, string>> {
+  const ssmPrefix = `/env/${env}/docs/worker_pool`;
+  const ssmClient = new SSMClient({ region: REGION });
+  const parameters = [
+    '/atlas/password',
+    '/atlas/username',
+    '/atlas/host',
+    '/atlas/dbname',
+    '/github/bot/password',
+  ];
+
+  const pathToEnvMap: Map<string, string> = new Map();
+
+  pathToEnvMap.set(`${ssmPrefix}/atlas/password`, 'MONGO_ATLAS_PASSWORD');
+  pathToEnvMap.set(`${ssmPrefix}/atlas/username`, 'MONGO_ATLAS_USERNAME');
+  pathToEnvMap.set(`${ssmPrefix}/atlas/host`, 'MONGO_ATLAS_HOST');
+  pathToEnvMap.set(`${ssmPrefix}/atlas/dbname`, 'MONGO_ATLAS_DBNAME');
+  pathToEnvMap.set(`${ssmPrefix}/github/bot/password`, 'GITHUB_BOT_PASSWORD');
+
+  const parametersMap: Record<string, string> = {};
+
+  await Promise.all(
+    parameters.map(async paramName => {
+      const fullParamPath = `${ssmPrefix}${paramName}`;
+      const getParamCommand = new GetParameterCommand({
+        Name: fullParamPath,
+        WithDecryption: true,
+      });
+
+      const envName = pathToEnvMap.get(fullParamPath);
+
+      try {
+        const ssmResponse = await ssmClient.send(getParamCommand);
+        const paramString = ssmResponse.Parameter?.Value;
+
+        if (!envName || !paramString) {
+          console.error(
+            `ERROR! Could not retrieve string for the following param: ${paramName}`,
+          );
+          return;
+        }
+
+        core.setSecret(paramString);
```

Comment:
This is incredibly convenient! I was worried about leaking secrets to the logs, but the GitHub Actions library has this function which obfuscates any string you want. It scans through the logs and replaces any occurrence of that string with `***`.